### ISSUE TITLE
min_max lint handles negative numbers incorrectly

### ISSUE THREAD

**Issue by Thinkofname** [rust-lang-nursery/rust-clippy#2741]
The code:
```rust
    let test = 554;
    println!("{}", cmp::min(cmp::max(test, -30_000), 30_000));
```
Gives the clippy error
```
error: this min/max combination leads to constant result
 --> src/main.rs:4:20
  |
4 |     println!("{}", cmp::min(cmp::max(test, -30_000), 30_000));
  |                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: #[deny(min_max)] on by default
  = help: for further information visit https://rust-lang-nursery.github.io/rust-clippy/v0.0.195/index.html#min_max
```
Yet running the code will produce the correct output. Changing the `-30_000` to `0` makes the clippy error stop.

Playground example: http://play.rust-lang.org/?gist=b365f1fc9b521abbeb05fbace8fb5b45&version=stable&mode=debug
